AngularJS/src/org/angular2/entities/source/Angular2SourceDirective.java

```diff
@@ -143,6 +161,51 @@ public boolean isTemplate() {
                 Collections.unmodifiableCollection(outputs.values()));
   }
 
+  @NotNull
+  private Angular2CtorBoundParameters getCtorParameters() {
+    // The @Attribute decorated parameters can be found (and are valid) only
+    // in the class constructor, thus we have to get hold of it and process each parameter
+    final JSFunction ctor = getTypeScriptClass().getConstructor();
```

Comment:
There can be multiple constructors in TS class, which one is valid?

Reply:
If there is only one constructor, I'll use that one. If there are multiple, I'll pick the only one which has `isOverloadImplementation == true`